src/librustc_mir/borrow_check.rs

```diff
@@ -396,26 +396,31 @@ impl<'c, 'b, 'a: 'b+'c, 'gcx, 'tcx: 'a> MirBorrowckCtxt<'c, 'b, 'a, 'gcx, 'tcx>
                             ReadKind::Copy =>
                                 this.report_use_while_mutably_borrowed(
                                     context, lvalue_span, borrow),
-                            ReadKind::Borrow(bk) =>
+                            ReadKind::Borrow(bk) =>{
+                                let end_issued_loan_span =
+                                    flow_state.borrows.base_results.operator().region_span(
+                                        &borrow.region).end_point();
                                 this.report_conflicting_borrow(
-                                    context, lvalue_span,
-                                    (lvalue_span.0, bk), (&borrow.lvalue, borrow.kind)),
+                                    context, lvalue_span, bk, &borrow, end_issued_loan_span)
+                            },
                         }
                         Control::Break
                     }
                     (Write(kind), _) => {
                         match kind {
-                            WriteKind::MutableBorrow(bk) =>
+                            WriteKind::MutableBorrow(bk) => {
+                                let end_issued_loan_span =
+                                    flow_state.borrows.base_results.operator().region_span(
+                                        &borrow.region).end_point();
                                 this.report_conflicting_borrow(
-                                    context, lvalue_span,
-                                    (lvalue_span.0, bk), (&borrow.lvalue, borrow.kind)),
+                                    context, lvalue_span, bk, &borrow, end_issued_loan_span)},
```

Comment:
nit: coding style in project is to both omit the trailing comma here (see earlier comment) and also put a line break before the ending `}`